numpy/core/src/multiarray/dtype_transfer.c

```diff
@@ -228,8 +217,8 @@ _strided_to_strided_any_to_object(
     }
     if (data->decref_src.func != NULL) {
         /* If necessary, clear the input buffer (`move_references`) */
-        if (data->decref_src.func(&data->decref_src.context,
-                &orig_src, &N, &src_stride, data->decref_src.auxdata) < 0) {
+        if (data->decref_src.func(NULL, data->decref_src.descr,
+                orig_src, N, src_stride, data->decref_src.auxdata) < 0) {
```

Comment:
Hmmm, it might be possible to hit this for structured to object (I suspect not, but I will just try to add a test).  And we only have object and structured, user dtypes could (in the future) hit it, though.